Aardvark/ARKEmailBugReporter.m

```diff
@@ -37,6 +37,82 @@ @interface ARKInvisibleView : UIView
 @end
 
 
+@interface CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+
+@end
+
+
+@implementation CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+{
+    NSMutableString *const mutableDescription = [NSMutableString new];
+    for (int i = 0 ; i < indentationLevel ; i++) {
+        [mutableDescription appendString:@"   | "];
+    }
+    
+    [mutableDescription appendFormat:@"%@\n", [self description]];
+    
+    for (CALayer *sublayer in self.sublayers) {
+        [mutableDescription appendString:[sublayer _recursiveLayerHierarchyDescriptionWithIndentationLevel:(indentationLevel + 1)]];
+    }
+    
+    return mutableDescription;
+}
+
+@end
+
+
+@interface UIView (HierarchyDescription)
+
+/// Re-implementation of `-[UIView recursiveDescription]` to avoid calling private API.
```

Comment:
The fact that this replicates SPI is somewhat orthogonal; I'd lead with a straightforward description of what it does, and maybe say parenthetically that it's like `recursiveDescription`. Also should have `ARK_` prefix.